src/sidebar/util/watch.js

```diff
@@ -1,81 +1,68 @@
-import shallowEqual from 'shallowequal';
-
 /**
- * Watch for changes of computed values.
+ * Watch a data source for changes to a subset of its data.
+ *
+ * `watch` subscribes to change notifications from a data source, computes
+ * values using the data in it, and runs a callback with the current and
+ * previous computed values each time the computed result changes.
  *
- * This utility is a shorthand for a common pattern for reacting to changes in
- * some data source:
+ * @example
+ *   const unsubscribe = watch(
+ *     store.subscribe,
  *
- * ```
- * let prevValue = getCurrentValue();
- * subscribe(() => {
- *   const newValue = getCurrentValue();
- *   if (prevValue !== newValue) {
- *     // Respond to change of value.
- *     // ...
+ *     // Extract some data of interest from the store.
+ *     () => store.getValue(),
  *
- *     // Update previous value.
- *     prevValue = new value;
- *   }
- * });
- * ```
+ *     // Use strict comparison of values
+ *     null,
```

Comment:
Looks like the argument ordering needs updating in these comments since you swapped `callback` and `compare` ordering.

Reply:
Ah, thanks for catching. Will fix.